File: backend-java/src/main/java/org/egualpam/contexts/observable/shared/application/domain/AggregateId.java
package org.egualpam.contexts.observable.shared.application.domain;

import org.egualpam.contexts.observable.shared.application.domain.exceptions.InvalidAggregateId;

import java.util.UUID;

public record AggregateId(String value) {
    public AggregateId {
        try {
            UUID.fromString(value);
        } catch (IllegalArgumentException e) {
            throw new InvalidAggregateId(value);
        }
    }
}

File: backend/src/main/kotlin/org/egualpam/contexts/observable/walletinteractionservice/shared/application/domain/AggregateId.kt
package org.egualpam.contexts.observable.walletinteractionservice.shared.application.domain

import org.egualpam.contexts.observable.walletinteractionservice.shared.application.domain.exceptions.InvalidAggregateId
import java.util.UUID

open class AggregateId(private val value: String) {
  init {
    try {
      UUID.fromString(value)
    } catch (e: IllegalArgumentException) {
      throw InvalidAggregateId(value)
    }
  }

  fun value() = value
}
